sdks/java/io/amazon-web-services2/src/test/java/org/apache/beam/sdk/io/aws2/common/ClientConfigurationTest.java

```diff
@@ -50,4 +52,28 @@ public void testSerialization() {
 
     assertThat(deserializedConfig).isEqualTo(config);
   }
+
+  @Test
+  public void testJsonSerialization() {
```

Comment:
Initially I planned to ignore RetryConfiguration when serializing to/from Json. While I don't see a use case yet, it might be misleading / error prone. So added that capability and respective tests ...

`regionId` is purely internal and tested through the respective public api using `region`

Reply:
@echauchot I've pushed that change

Reply:
:+1: